### ISSUE TITLE
Model transfer time is too long

### ISSUE THREAD

**Issue by XiaotianJia** [OpenMined/PySyft#4028]
I use syft to transport the yolov3 network between the client and the server (the side with data). I have some questions about the network transmission time. The following is part of my code.

`async def fit_model_on_worker(
    worker: websocket_client.WebsocketClientWorker,
    traced_model: torch.jit.ScriptModule,
    batch_size: int,
    curr_round: int,
    max_nr_batches: int,
    lr: float,
):
    train_config = sy.TrainConfig(
        model=traced_model,
        loss_fn=loss_unet,
        batch_size=batch_size,
        shuffle=True,
        max_nr_batches=max_nr_batches,
        epochs=1,
        optimizer="SGD",
        optimizer_args={"lr": lr},
    )

    train_config.send(worker)
    loss = await worker.async_fit(dataset_key="mnist", return_ids=[0], device="cuda")
    model = train_config.model_ptr.get().obj
`
The running time of `train_config.send(worker)` and `model = train_config.model_ptr.get().obj
` is too long. The `.pth` file of the yolov3 network is 236 MB. But it takes nearly an hour and a half for one transmission.

Another question is if there are `n` servers, the network needs to be send to the `n` servers and the sending time will be n times. Can `train_config.send(worker)`  be asynchronous? Is there any way to shorten the transmission time of the model?

**Comment by IonesioJunior** [comment]
Hello @XiaotianJia! ;)
Answering the first question:  We have some limitations due to the use of dependencies that don't optimize the transfer rate of large files. We are working on possible solutions for this and searching by optimized libraries. One of those limitations was solved and explained in this [PR ](https://github.com/OpenMined/PyGrid/pull/689) and was merged in PyGrid's dev branch.

Answering the second question: Yep, In the next PySyft version (0.3.0). I'll probably adopt async coroutines as a default approach to send/receive messages using WebSockets. You can learn about it by checking this [PR](https://github.com/OpenMined/PySyft/pull/4009/files#).


If you have any other questions/issues/suggestions feel free to create new issues or ask us in our slack channel #lib_pygrid.
